#TechTrendz Retail Business Analysis Report
A Comprehensive Case Study Using NumPy

\

Author: Rashi Ratan \
Published: February 10, 2024

\

TechTrendz Retail is a rapidly growing company that recently launched a new product line. As a Business Analytics intern at the company, I am tasked with creating a detailed analysis of the company's performance in three key areas:

1. Sales Analysis: Evaluate the daily sales data by converting raw data into a NumPy array, applying a 10% discount, and computing aggregate statistics such as total, average, and standard deviation of the discounted sales.

2. Inventory and Discount Analysis: Analyze a two-dimensional dataset that contains inventory counts and corresponding discount percentages for various product categories. Extract relevant columns, compute the total inventory, and use advanced slicing to isolate data for selected product groups.

3. Sales Trend Analysis: Investigate how sales evolve over a week. Compute the cumulative sales and day-to-day differences, then identify days with significant increases in sales.

##Problem 1: Sales Analysis

To analyzie the sales, I first imported the .csv file as a NumPy array. After this, I used a vectorized operation to calculated discounted sales array, using which I calculate key metrics.

In [8]:
#import numpy
import numpy as np

#make sure to upload the files for each runtime
#converting .csv file to NumPy array
sales_csv = np.loadtxt("/content/drive/MyDrive/Colab Notebooks/sales.csv", delimiter=",", skiprows=0)

sales = np.array(sales_csv)
print(f' Sales Array: {sales} \n')
#print to verify correct importing of data

#calculating discounted sales
discounted_sales = sales*0.9
print(f"Discounted Sales: {discounted_sales} \n")

#calculate sum of discounted sales
sum_discounted_sales = np.sum(discounted_sales)
print(f"Total Discounted Sales: {sum_discounted_sales:.2f} \n")

#calculate average of discounted sales
avg_discounted_sales = np.mean(discounted_sales)
print(f"Average Discounted Sales: {avg_discounted_sales:.2f} \n")

#calculate standard deviation of discounted sales
std_discounted_sales = np.std(discounted_sales)
print(f"Standard Deviation of Discounted Sales: {std_discounted_sales:.2f} \n")


 Sales Array: [100. 150. 220. 280. 340. 400. 470.] 

Discounted Sales: [ 90. 135. 198. 252. 306. 360. 423.] 

Total Discounted Sales: 1764.00 

Average Discounted Sales: 252.00 

Standard Deviation of Discounted Sales: 111.27 



###Report:
Discounted Sales for the week are [90, 135, 198, 252, 306, 360, 423]. The total sales for the week are 1764 units. Average discounted sales per day amount to about 252 units per day, with a standard deviation of about 111 units. Therefore, there seems to be a considerable amount of variability in the sales everyday. This sales data would be helpful in spotting anomalies and diving deeper into the factors that may be influencing them.


##Problem 2: Inventory and Discount Analysis

After importing the data from the .csv file into an array, I analyzed the inventory and discount information by extracting the relevant columns and computing the total inventory. I also isolated the data for the first two product categories from the data array.

In [9]:
#make sure to upload the files for each runtime
#importing inventory data from .csv file
inventory_csv = np.loadtxt("/content/drive/MyDrive/Colab Notebooks/inventory.csv", delimiter=',', skiprows=0, dtype=int)

# Inventory and discount data (rows: product categories, columns: [inventory, discount])
data = np.array(inventory_csv)
print(f'Inventory Data: {data} \n')
#print inventory to verify correct importing of data

#extracting inventory counts and discounts
inventory = data[:, 0]
print(f'Inventories for all Product Categories: {inventory}')

discount = data[:, 1]
print(f'Discounts for all product categories: {discount}\n')

#computing total inventory
total_inventory = int(np.sum(inventory))
print(f'Total Inventory: {total_inventory} units')

#submatrix for first two product categories
submatrix = data[:2, :]
print(f'Submatrix for first two product categories:\n {submatrix}\n')


Inventory Data: [[100  10]
 [150  15]
 [200  20]
 [250  25]
 [300  30]] 

Inventories for all Product Categories: [100 150 200 250 300]
Discounts for all product categories: [10 15 20 25 30]

Total Inventory: 1000 units
Submatrix for first two product categories:
 [[100  10]
 [150  15]]



###Report:
The inventory counts for all the product categories are [100, 150, 200, 250, 300] with their corresponding discounts of [10, 15, 20, 25, 30]. The total inventory is 1000 units across all product categories. The data for the first two product categories is:
```
 [[100  10]
 [150  15]]
```
An understanding of this is very important to be able to determine the influence of discounts on inventory turnover metrics.





##Problem 3: Sales Trend Analysis

To assess the sales trend, I computed the cumulative sales as well as the daily sales difference. Then, I filtered the differences to pinpoint days with an increase of more than 50 units.

In [3]:
#display daily sales of the week
daily_sales = sales
print(f'Daily Sales: {daily_sales} \n')

#calculate cumulative sales for the week
cumulative_sales = np.cumsum(sales)
print(f'Cumulative Sales for the Week: {cumulative_sales}\n')

#calculate day-to-day differences in sales
sales_diff = np.diff(sales)
print(f'Day-to-day Differences in Sales: {sales_diff}\n')

# Filter for significant sales increases (difference > 50 units).
significant_increase = sales_diff[sales_diff > 50]
print(f'Significant Sales Increases (diff > 50): {significant_increase}\n')

Daily Sales: [100. 150. 220. 280. 340. 400. 470.] 

Cumulative Sales for the Week: [ 100.  250.  470.  750. 1090. 1490. 1960.]

Day-to-day Differences in Sales: [50. 70. 60. 60. 60. 70.]

Significant Sales Increases (diff > 50): [70. 60. 60. 60. 70.]



###Report:
The cumulative sales for the week were: [100, 250, 470, 750, 1090, 1490, 1960]. The day-to-day differences in sales were: [50, 70, 60, 60, 60, 70]. Significant increases in sales, of more than 50 units, are observed for multiple days. This helps to identify periods of peak or better performances and help in formulation of promotion strategies.

##Conclusion:

Using NumPy, this report has addressed key business areas of TechTrendz Retail. By analyzing the sales data, inventory information as well as sales trends, the report has made observations and developed actionable insights for the business to inform and support decision-making. The results indicate that sales in the company are moderately stable, but not stagnant. There have been significant increases on multiple days, showing even growth in the sales numbers. This would help with determining the inventory requirements for TechTrendz. Further investigation into the factors driving these trends is recommended.

\


*Rashi Ratan* \
*Business Analytics Intern, TechTrendz Retail* \
*Dr. Tim C. Smith ©️2025*